pkg/controller/download_controller.go

```diff
@@ -297,6 +298,34 @@ func (c *downloadController) patchDownload(req *pluginv1api.Download, mutate fun
 	return req, nil
 }
 
+func (c *downloadController) patchDownloadByStatusWithRetry(req *pluginv1api.Download, newPhase pluginv1api.DownloadPhase, msg string) (*pluginv1api.Download, error) {
+	var updatedDownload *pluginv1api.Download
+	var err error
+	var retryCnt = 0
+	log := loggerForDownload(c.logger, req)
+	log.Info("Ready to call patchDownloadByStatus API. Will retry on network issue for %d times.", utils.RetryMaximum)
+	err = wait.PollImmediate(utils.RetryInterval * time.Second, utils.RetryInterval * utils.RetryMaximum * time.Second, func() (bool, error) {
```

Comment:
Taking advantage of `wait.PollImmediate`, we don't actually have to explictly record the `retryCnt`.
In `wait.PollImmediate`, the first argument specify the retry interval and the second one specify the time-out. In the case of retrying on patch failure, e.g.
```
err = wait.PollImmediate(utils.RetryIntervalInSeconds * time.Second, utils.RetryTimeoutInSeconds * time.Second, func() (bool, error) {
    if _, err = c.patchDownloadByStatus(req, newPhase, msg); err != nil {
          // you can add logging here
          return false, nil
    }
    return true, nil
})
...
```
Please double check whether it fit in your case.